waspc/src/Wasp/Analyzer/Parser/Parser.y

```diff
@@ -25,6 +26,8 @@ import Control.Monad.Except (throwError)
 --  - to call `parseError` when the parser encounters an error
 --  - to provide `parseError` with list of expected tokens that would avoid the error
 %name parse
+%name parseExpression Expr
```

Comment:
It turns out this is all we need to get a function that parses with an arbitrary non-terminal symbol.

Reply:
Ha this is awesome! I wasn't aware we can produce multiple parser functions like this!

What about `parse` above -> it by default starts from the first rule in grammar (`Stmt`)? Or it is able to figure out the top-level rule of grammar in some smart way?

Reply:
Yes, it starts from the first one by default, but I've made some changes when cleaning it up (hint: cabal philosophy).

Reply:
hah I guess you made it more explicit :D? I like that!